lib/index.ts

```diff
@@ -21,116 +21,105 @@ limitations under the License.
 import * as url from 'url';
 
 import * as errors from 'balena-errors';
-import * as Promise from 'bluebird';
-import { PinejsClientCoreFactory } from 'pinejs-client-core';
+import { AnyObject, Params, PinejsClientCore } from 'pinejs-client-core';
 
-const PinejsClientCore = PinejsClientCoreFactory(Promise);
-
-function getPine(param: {
+interface BackendParams {
 	apiUrl: string;
 	apiVersion: string;
 	apiKey?: string;
 	request: {
 		// TODO: Should be the type of balena-request
-		send: (
-			options: PinejsClientCoreFactory.AnyObject,
-		) => Promise<{ body: any }>;
+		send: (options: AnyObject) => Promise<{ body: any }>;
 	};
 	auth: import('balena-auth').default;
-}): getPine.BalenaPine {
-	const { apiUrl, apiVersion, apiKey, request, auth } = param;
-	const apiPrefix = url.resolve(apiUrl, `/${apiVersion}/`);
+}
+
+/**
+ * @class
+ * @classdesc A PineJS Client subclass to communicate with balena.
+ * @private
+ *
+ * @description
+ * This subclass makes use of the [balena-request](https://github.com/balena-io-modules/balena-request) project.
+ */
+class BalenaPine extends PinejsClientCore<BalenaPine> {
+	public backendParams: BackendParams;
+	public API_URL: string;
+	public API_VERSION: string;
+	public API_PREFIX: string;
+
+	constructor(params: Params, backendParams: BackendParams) {
+		const { apiUrl, apiVersion } = backendParams;
+		const apiPrefix = url.resolve(apiUrl, `/${apiVersion}/`);
+		super({
+			...params,
+			apiPrefix,
+		});
+
+		// TS doesn't allow to declare `public backendParams: BackendParams` in the
+		// constructor, since we need to run code before the super() call.
+		this.backendParams = backendParams;
+		this.API_URL = apiUrl;
+		this.API_VERSION = apiVersion;
+		this.API_PREFIX = apiPrefix;
```

Comment:
This is a duplicate of the existing `PinejsClientCore.apiPrefix` property btw, I'm ok with keeping it but it might be worth thinking about moving to the more standard one